src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.hotspot/src/jdk/vm/ci/hotspot/HotSpotJVMCIBackendFactory.java

```diff
@@ -32,4 +39,45 @@
      * Gets the CPU architecture of this backend.
      */
     String getArchitecture();
+
+    /**
+     * Converts a bit mask of CPU features to enum constants.
+     *
+     * @param <CPUFeatureType> CPU feature enum type
+     * @param enumType the class of {@code CPUFeatureType}
+     * @param constants VM constants. Each entry whose key starts with {@code "VM_Version::CPU_"}
+     *            specifies a CPU feature and its value is a mask for a bit in {@code features}
+     * @param features bits specifying CPU features
+     * @param renaming maps from VM feature names to enum constant names where the two differ
+     * @throws IllegalArgumentException if any VM CPU feature constant cannot be converted to an
+     *             enum value
+     * @return the set of converted values
+     */
+    static <CPUFeatureType extends Enum<CPUFeatureType>> EnumSet<CPUFeatureType> convertFeatures(
+                    Class<CPUFeatureType> enumType,
+                    Map<String, Long> constants,
+                    long features,
+                    Map<String, String> renaming) {
+        EnumSet<CPUFeatureType> outFeatures = EnumSet.noneOf(enumType);
+        List<String> missing = new ArrayList<>();
+        for (Entry<String, Long> e : constants.entrySet()) {
+            long bitMask = e.getValue();
+            String key = e.getKey();
+            if (key.startsWith("VM_Version::CPU_")) {
```

Comment:
As I understand this code, it goes over `constants` values passed from VM and Trying to map them to `enumType`. It catches cases when a value is missing in `enumType`. What about case when `enumType` has `extra` value which is not defined in `constants`?

Reply:
Okay.